python/lsst/daf/butler/datastores/posixDatastore.py

```diff
@@ -222,20 +222,55 @@ def _removeFileExists(path):
         info = self._extractIngestInfo(path, ref, formatter=formatter)
         self._register_datasets([(ref, info)])
 
+    def _overrideTransferMode(self, *datasets: FileDataset, transfer: Optional[str] = None) -> str:
+        # Docstring inherited from base class
+        if transfer != "auto":
+            return transfer
+        try:
+            # Assume first dataset is representative
+            self._pathInStore(datasets[0].path)
+        except RuntimeError:
+            transfer = "link"
+        else:
+            transfer = None
+        return transfer
+
+    def _pathInStore(self, path: str) -> str:
+        """Return path relative to datastore root
+
+        Parameters
+        ----------
+        path : `str`
+            Path to dataset. Can be absolute path. Returns path in datastore
+            or raises an exception if the path it outside.
+
+        Returns
+        -------
+        inStore : `str`
+            Path relative to datastore root.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if the supplied path is outside the datastore.
```

Comment:
Given that in at least one context this is a totally normal condition (as opposed to an error) it might be better to return `None`.